components/util.py

```diff
@@ -64,45 +66,80 @@ def build_menu(
 
 
 def rate_limit_tracker(_: Update, context: CallbackContext) -> None:
-    data = cast(Dict, context.chat_data).setdefault('rate_limit', {})
+    data = cast(Dict, context.chat_data).setdefault("rate_limit", {})
 
     for key in data.keys():
         data[key] += 1
 
 
-Func = TypeVar('Func', bound=Callable[[Update, CallbackContext], None])
+Func = TypeVar("Func", bound=Callable[[Update, CallbackContext], None])
 
 
 def rate_limit(
     func: Callable[[Update, CallbackContext], None]
 ) -> Callable[[Update, CallbackContext], None]:
     """
     Rate limit command so that RATE_LIMIT_SPACING non-command messages are
-    required between invocations.
+    required between invocations. Private chats are not rate limited.
     """
 
     @wraps(func)
     def wrapper(update: Update, context: CallbackContext) -> None:
+        if chat := update.effective_chat:
+            if chat.type == chat.PRIVATE:
+                return func(update, context)
+
         # Get rate limit data
-        try:
-            data = cast(Dict, context.chat_data)['rate_limit']
-        except KeyError:
-            data = cast(Dict, context.chat_data)['rate_limit'] = {}
+        data = cast(Dict, context.chat_data).setdefault("rate_limit", {})
 
         # If we have not seen two non-command messages since last of type `f`
         if data.get(func, RATE_LIMIT_SPACING) < RATE_LIMIT_SPACING:
-            logging.debug('Ignoring due to rate limit!')
+            logging.debug("Ignoring due to rate limit!")
+            cast(Message, update.effective_message).delete()
             return None
 
         data[func] = 0
-
         return func(update, context)
 
     return wrapper
 
 
 def truncate_str(string: str, max_length: int) -> str:
-    return (string[:max_length] + '…') if len(string) > max_length else string
+    return (string[:max_length] + "…") if len(string) > max_length else string
+
+
+def build_command_list(
+    private: bool = False, group_name: str = None, admins: bool = False
+) -> List[Tuple[str, str]]:
+
+    base_commands = [
+        ("docs", "Send the link to the docs."),
+        ("wiki", "Send the link to the wiki."),
+        ("help", "Send the link to this bots README."),
+    ]
+    hint_commands = [(hint.tag, hint.description) for hint in TAG_HINTS.values()]
+
+    if private:
+        return base_commands + hint_commands
+
+    base_commands += [("rules", "Show the rules for this group.")]
+
+    if not group_name:
```

Comment:
```suggestion
    if group_name is None:
```
this is little more clear